### ISSUE TITLE
Python package identification might throw exception on valid setup.py/setup.cfg

### ISSUE THREAD

**Issue by murphm8** [colcon/colcon-core#131]
Pip is adding support for PEP 508 URL dependencies pypa/pip#5571. I believe that the code linked below will throw an exception if there is a URL in the dependencies list since it is 'not a valid version'.

https://github.com/colcon/colcon-core/blob/83bf0391323514ab8ada7aacc693fd16f59c1037/colcon_core/package_identification/python.py#L125

**Comment by dirk-thomas** [comment]
Can you please provide a reproducible example.

**Comment by murphm8** [comment]
Hey, sorry I haven't followed up. Been super busy. I will get you a reproduction as soon as I am able.

**Comment by murphm8** [comment]
I ran a quick test. I must've been using an improperly formatted string when I tried this before. I tested building a python package with a dependency of `colcon-core@git+https://github.com/colcon/colcon-core.git#egg=colcon-core` and `colcon-core@git+https://github.com/colcon/colcon-core.git@0.3.12#egg=colcon-core-0.3.12`. 

The `create_dependency_descriptor` returned a DependencyDescriptor with name `colcon-core`. For the dependency with the version specified there is no version metadata, but I think that is expected.

For bundle I am installing dependencies into the bundle based on this metadata. It might be beneficial to put these PEP 508 urls in the metadata so the correct version is installed when the dependency was specified as a PEP 508 url.

Closing this since it doesn't appear that the parse requirement will fail with a correctly formatted PEP 508 url.
